### ISSUE TITLE
Better support for plotting text in 3D at different heights

### ISSUE THREAD

**Issue by weiji14** [GenericMappingTools/pygmt#1393]
**Description of the desired feature**

For 3D plots, it should be possible to plot text at different heights based on a 'z' column, see https://docs.generic-mapping-tools.org/6.2/text.html#z. The current implementation of `text` in PyGMT (see https://www.pygmt.org/v0.4.0/api/generated/pygmt.Figure.text.html) does support this, but users have to provide the xyz input into the `textfiles` parameter, and it is also not very discoverable for new users.

It would be nice to have a `z` parameter (in addition to `x` and `y`) to be clear that `text` does support multi-height plotting. Here's a checklist for anyone wanting to work on this 'enhancement'.

TODO:
1. Alias the [`-Z`](https://docs.generic-mapping-tools.org/6.2/text.html#z), call it `zvalue` (as in `plot3d`)?
2. Add the `z` parameter here:
    https://github.com/GenericMappingTools/pygmt/blob/945bbc57ffc89a7c29384df48a653295b3f1a435/pygmt/src/text.py#L48-L49
3. Modify the code here to mention that x/y/z triples are also possible (need to wait for #1121 though):
    https://github.com/GenericMappingTools/pygmt/blob/945bbc57ffc89a7c29384df48a653295b3f1a435/pygmt/src/text.py#L160-L170
4. Update the `text` API docstring to document the new behaviour.
5. (Optional) Update the tutorial at https://www.pygmt.org/v0.4.0/tutorials/text.html to mention that z-value inputs are supported too.

**Are you willing to help implement and maintain this feature?** Please wait for #1121 before starting work on this.

**Comment by landryza** [comment]
Hi @weiji14 I would like to work on this issue. The description of what is needed is straightforward and clear. I just want to check if there are any helper functions that will be impacted by these changes.

If this feature is still required, I would be happy to have it assigned to me and help!

**Comment by weiji14** [comment]
Hi @landryza, please go ahead and give this a try! The underlying API has changed a bit since 5 years ago, but hopefully you can still figure out how to insert the `z` parameter. Let us know if you need any help :D
